tests/test_unit/test_kinematics.py

```diff
@@ -518,20 +521,40 @@ def test_compute_forward_vector_with_invalid_input(
 
 
 def test_nan_behavior_forward_vector(
-    valid_data_array_for_forward_vector_with_nans,
+    valid_data_array_for_forward_vector_with_nans: xr.DataArray,
 ):
     """Test that ``compute_forward_vector()`` generates the
     expected output for a valid input DataArray containing ``NaN``
     position values at a single time (``1``) and keypoint
     (``left_ear``).
     """
+    nan_time = 1
     forward_vector = kinematics.compute_forward_vector(
         valid_data_array_for_forward_vector_with_nans, "left_ear", "right_ear"
     )
-    assert (
-        np.isnan(forward_vector.values[1, 0, :]).all()
-        and not np.isnan(forward_vector.values[[0, 2, 3], 0, :]).any()
-    )
+    # Check coord preservation
+    for preserved_coord in ["time", "space", "individuals"]:
+        assert np.all(
+            forward_vector[preserved_coord]
+            == valid_data_array_for_forward_vector_with_nans[preserved_coord]
+        )
+    assert set(forward_vector["space"].values) == {"x", "y"}
```

Comment:
FYI, there is almost a verbatim repetition between this code block and the one added to `test_compute_forward_vector`. That said, I don't advocate for absolute adherence to the DRY principle, and in this case the refactoring effort required to remove the duplication may not be worth it.

Feel free to ingore this comment.

Reply:
Probably worth considering if we add another test that needs this check (I tend to live by 3 repeats -> DRY applies for short code blocks :sweat_smile:). Since it's just these two tests (for now) I'll leave as is, but will keep it in mind in the future.